src/components/picker/helpers/usePickerMode.ts

```diff
@@ -0,0 +1,47 @@
+import {ExpandableOverlayProps} from 'src/incubator';
+import {PickerProps, PickerModeTypes} from '../index';
+import {CustomPickerProps} from '../types';
+
+type PickerType = {
+  dialog?: boolean;
+  wheelPicker?: boolean;
+  modal?: boolean;
+  custom?: boolean;
+};
+
+const usePickerMode = (props: PickerProps) => {
+  let type: PickerType = {dialog: false, wheelPicker: false, modal: false, custom: false};
+  let headerProps: any;
+  let renderCustomModal: CustomPickerProps['renderCustomModal'];
+  let dialogProps: ExpandableOverlayProps['dialogProps'];
+  let pickerModalProps: ExpandableOverlayProps['modalProps'];
+  if ('pickerType' in props) {
```

Comment:
What this check is for?

Reply:
We can change it and extract the `pickerType` outside since it will always be passed.
I tried to make some using of conditional typing that didn't worked.

Reply:
I see you're checking if(pickerType) instead so this is good